src/components/CartAppContainer.jsx

```diff
@@ -0,0 +1,50 @@
+import {useState, useEffect} from 'react';
+import FETCH_DATA from '../data';
+import NavBar from './NavBar';
+import ProductTable from './ProductTable';
+
+const CartAppContainer = (props) => {
+  const [cart, setCart] = useState(0);
+  const [products, setProducts] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchData = async () => {
+    const response = await FETCH_DATA().then(res => res);
+    if (response) {
+      setProducts(response);
+      setIsLoading(false);
+    }
+  }
+
+  const updateCartCount = () => {
+    // make a copy of array held in PRODUCTS data blob using spread syntax
+    const newProducts = [...products];
+    // grab all products that have a count of at least 1
+    const productsWithCount = newProducts.filter(product => product.count > 0);
+    // initialize a variable to keep track of total products in cart, 
+    // add up all current product count and update 'cart' state variable with the sum
+    let totalProductsInCart = 0;
+    productsWithCount.forEach(product => {
+      totalProductsInCart += product.count;
+    });
+    setCart(totalProductsInCart);
+  }
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  return (
+    <>
+      <NavBar cart={cart}/>
+      <div>
+        {isLoading ? 
```

Comment:
I'd catch the loading state prior to the return statement to keep it clearer. 
i.e. 
```
if (!products) // or if (isLoading) {
  return <p>Loading...</p>
}

return (
  <ProductTable ... />
)
```